### PR TITLE
stats: completions and corrections

### LINKED ISSUE #2847: Billing statistics: completion, correction
## User story

As a service provider, I want to access the billing statistics so that I can invoice the service.

## Acceptance Criteria

* The ILL requests are added in the billing statistics
    * Request of the 3 last months (or same time range than the loans and other indicators) are considered AND with status pending|closed|validated. See request below:     
/api/ill\_requests/?q=\_created:%5B2022-01-01%20TO%202022-03-31%5D%20AND%20status:(closed%20OR%20pending%20OR%20validated)&simple=0https://bib.rero.ch/api/ill\_requests/?q=\_created:%5B2022-01-01%20TO%202022-03-31%5D%20AND%20status:(closed%20OR%20pending%20OR%20validated)&page=1&size=10&library=38&simple=0

* The indicator “Number of librarians” (n-opr) is corrected to include all librarians, whatever their role (currently, system librarian and acquisitions librarian are not counted).

## For later

* harmonisation billing stat vs. system librarian stat.
* access to billing stat by system librarian (preceding point must be solved)
* access to stat from professional interface instead of public interface

### PR DESCRIPTION
**Pull request by rerowep**
* Corrects `number_of_satisfied_ill_request`
  with status pending|closed|validated.
* Corrects "Number of librarians" to include all librarians.
* Closes #2847 

Co-Authored-by: Peter Weber <peter.weber@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by lauren-d — APPROVED** [review]

**Review by vgranata — APPROVED** [review]

**Review by benerken — APPROVED** [review]

File: rero_ils/modules/stats/api.py [hunk 1]
```diff
@@ -232,23 +244,19 @@ def number_of_circ_operations(self, library_pid, trigger):
...
-        for res in RecordsSearch(index=LoanOperationLog.index_name)\
-                .filter('range', date=self.date_range)\
-                .filter('term', record__type='illr')\
-                .filter('term', library__pid=library_pid)\
-                .filter('term', ill_request__status='validated')\
-                .source(['record'])\
-                .scan():
-            requests.add(res.record.pid)
-        return len(requests)
+        query = ILLRequestsSearch()\
```
Comment (jma) [inline rero_ils/modules/stats/api.py:255]: The problem here is that if a request is deleted it is not in stats.

File: rero_ils/alembic/21a994dc2beb_add_library_pid_to_ill_request_.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    used to calculate fees based on checkout library.
+    """
+    query = RecordsSearch(index=LoanOperationLog.index_name)\
+        .filter('term', record__type='illr')
+    pids = [hit.pid for hit in query.source('pid').scan()]
+    LOGGER.info(f'Upgrade operation logs illr :: {len(pids)}')
+    errors = 0
+    for idx, pid in enumerate(pids, 1):
+        record = LoanOperationLog.get_record(pid)
+        ill_request = ILLRequest.get_record_by_pid(record['record']['value'])
```
Comment (jma) [inline rero_ils/alembic/21a994dc2beb_add_library_pid_to_ill_request_.py:47]: What's happens if the Ill request has been deleted?

File: rero_ils/modules/operation_logs/jsonschemas/operation_logs/operation_log-v0.0.1.json [hunk 3]
```diff
@@ -117,6 +117,11 @@
             "ITEM_ON_LOAN",
             "ITEM_RETURNED"
           ]
+        },
+        "library_pid": {
+          "title": "Library PID",
+          "type": "string",
```
Comment (jma) [inline rero_ils/modules/operation_logs/jsonschemas/operation_logs/operation_log-v0.0.1.json:123]: why not type `keyword`?

**Review by jma — CHANGES_REQUESTED** [review]

File: rero_ils/modules/operation_logs/jsonschemas/operation_logs/operation_log-v0.0.1.json (hunk 3 above)
Comment (rerowep) [reply rero_ils/modules/operation_logs/jsonschemas/operation_logs/operation_log-v0.0.1.json:123]: All other `library_pid` are strings too in the mapping.

**Review by jma — APPROVED** [review]

**Pull request merged**